### PR TITLE
Auto add classifiers

### LINKED ISSUE #64: Convert to `go mod` instead of `dep`
Should we convert dependencies to [go modules](https://github.com/golang/go/wiki/Modules) as a replacement for `dep`?

`go modules` is currently experimental and only available in `go 1.11`

### PR DESCRIPTION
**Pull request by necrophonic**
# Motivation and Context

Currently when creating a new survey, classifiers need to be manually added separately. This is variously been achieved with manual database inserts (bad) or using the `/surveys/{surveyId}/classifiers` [POST] endpoint (better, but still manual).

The classifiers for a survey are (currently) always the same, so it makes sense to just generate them when the survey is created.

# What has changed

The `/surveys` [POST] has been updated to create the classifiers as well as creating the core survey.
The `/surveys/{surveyId}/classifiers` [POST] endpoint still exists and functionality is unchanged.

Additionally the `/vendor` folder has been removed and the service upgraded to use `go modules` instead (dependencies now pinned using `go.mod` and `go.sum`). This means a change to the build/release job in concourse to call `go get`

Build step in concourse becomes simply:
```shell
  cd survey-service-pre-release-source
  go get ./...
  make
```

A separate pull request on https://github.com/ONSdigital/ras-deploy will be required.

# How to test?

- Spin up a local environment and create a new survey via the `/create` page of `response-operations-ui`.
- Check the resultant records in the database to see the classifiers have been created and linked to the survey.

# Links

- Fixes #64 to move away from vendor to go modules
- Associated PR: https://github.com/ONSdigital/ras-deploy/pull/110

### CODE REVIEW

**Comment by necrophonic** [comment]
When this PR goes in, the instructions on confluence for "how to create a new survey" should be updated to reflect the fact you no longer need to manually create the classifiers.

**Comment by necrophonic** [comment]
@ryangrundy7 Shouldn't have any issues with the Go version in the pipeline as it's using a binary buildpack to push to cloud foundry. The references to the go version and package in the old env vars would only be used if using the Go buildpack.

What errors are you seeing in the deploy job? (Which job/pipeline?)

File: models/surveys.go [hunk 1]
```diff
@@ -1,18 +1,23 @@
 package models
 
+// TODO All transactions should have rollbacks in the case of errors
```
Comment (jonnyshaw89) [inline models/surveys.go]: Is there any reason why these TODOs are not just done as part of this PR, as opposed to adding tech debt?

Comment (necrophonic) [reply models/surveys.go]: It's a good question :) It's not technical debt added by this PR (all the bits touched by the changes here have had that done). It was more an observation on the existing code that needs to be fixed at some point (outside the scope of this change). 
A card has been raised in the backlog to look at this.

Comment (jonnyshaw89) [reply models/surveys.go]: ah ok, no boy scout rule here then.

**Pull request merged**